http/mux.go

```diff
@@ -59,49 +57,33 @@ type (
 		Use(func(http.Handler) http.Handler)
 	}
 
-	// mux is the default Muxer implementation. It leverages the
-	// httptreemux router and simply substitutes the syntax used to define
-	// wildcards from ":wildcard" and "*wildcard" to "{wildcard}" and
-	// "{*wildcard}" respectively.
-	mux struct {
-		*httptreemux.ContextMux
+	// muxer is the default Muxer implementation.
+	muxer struct {
+		*mux.Router
 	}
 )
 
-// NewMuxer returns a Muxer implementation based on the httptreemux router.
+// NewMuxer returns a Muxer implementation based on Gorilla mux.
 func NewMuxer() MiddlewareMuxer {
-	r := httptreemux.NewContextMux()
-	r.EscapeAddedRoutes = true
-	r.NotFoundHandler = func(w http.ResponseWriter, req *http.Request) {
-		ctx := context.WithValue(req.Context(), AcceptTypeKey, req.Header.Get("Accept"))
-		enc := ResponseEncoder(ctx, w)
-		w.WriteHeader(http.StatusNotFound)
-		enc.Encode(NewErrorResponse(ctx, fmt.Errorf("404 page not found")))
-	}
-	return &mux{r}
+	r := mux.NewRouter()
```

Comment:
It needs `NotFoundHandler`, right?

```suggestion
	r := mux.NewRouter()
	r.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
		ctx := context.WithValue(req.Context(), AcceptTypeKey, req.Header.Get("Accept"))
		enc := ResponseEncoder(ctx, w)
		w.WriteHeader(http.StatusNotFound)
		enc.Encode(NewErrorResponse(ctx, fmt.Errorf("404 page not found")))
	})
```